homeassistant/components/stream/core.py

```diff
@@ -498,12 +506,13 @@ def _generate_image(self, width: int | None, height: int | None) -> None:
         if frames:
             frame = frames[0]
             if width and height:
-                if self._stream_settings.orientation >= 5:
+                if self._dynamic_stream_settings.orientation >= 5:
```

Comment:
This comparison is easier to see as an int. It happens that orientations 5, 6, 7, and 8 are rotated once so have flipped dimensions.